### PR TITLE
session_test, plan_test: move out some tests.

### PR DESCRIPTION
**Pull request by hanfei1991**
move out some nagetive cases and simplify them.

### CODE REVIEW

**Comment by shenli** [comment]
LGTM

**Comment by ngaut** [comment]
Nice clean up.

**Comment by shenli** [comment]
@coocood PTAL

File: plan/logical_plan_test.go [hunk 1]
```diff
@@ -1101,6 +1101,26 @@ func (s *testPlanSuite) TestValidate(c *C) {
...
+			err: ErrAmbiguous,
+		},
+		{
+			sql: "select a + 1 from t having a",
+			err: ErrUnknownColumn,
+		},
+		{
+			sql: "select a from t having sum(avg(a))",
+			err: ErrInvalidGroupFuncUse,
+		},
```
Comment (coocood) [inline plan/logical_plan_test.go]: The number of deleted cases is 12 but only added 5 cases.

Comment (hanfei1991) [reply plan/logical_plan_test.go]: some cases are duplicate

**Comment by coocood** [comment]
LGTM

**Pull request merged**